### PR TITLE
Remove some usages of `RVD.rdd`

### PR DESCRIPTION
**Pull request by danking**
These are some easy to remove references to `RVD.rdd`. I added `treeReduce`, which depends on `treeAggregate` (#3310) which will hopefully be merged soon. In the meantime, I've added the `treeAggregate` methods here so this PR will pass its tests.

### CODE REVIEW

**Comment by danking** [comment]
Changing assignee in order to more equitably distribute these PRs

File: src/main/scala/is/hail/table/Table.scala [hunk 1]
```diff
@@ -102,13 +103,71 @@ object Table {
...
+  ): Table = apply(hc, rdd, signature, Array.empty[String])
+
+  def apply(
+    hc: HailContext,
+    rdd: RDD[Row],
+    signature: TStruct,
+    key: IndexedSeq[String]
+  ): Table = apply(hc, rdd, signature, key, TStruct.empty())
+
+  def apply(
```
Comment (catoverdrive) [inline src/main/scala/is/hail/table/Table.scala]: This seems kind of weird---being able to specify an arbitrary globalSignature but the globals are always an empty struct?

File: src/main/scala/is/hail/table/Table.scala [hunk 2]
```diff
@@ -102,13 +103,71 @@ object Table {
...
+    globals)
+
+  def apply(
+    hc: HailContext,
+    crdd: ContextRDD[RVDContext, Row],
+    signature: TStruct,
+    key: IndexedSeq[String]
+  ): Table = apply(hc, crdd, signature, key, TStruct.empty())
+
+  def apply(
```
Comment (catoverdrive) [inline src/main/scala/is/hail/table/Table.scala]: ditto here.

File: src/main/scala/is/hail/table/Table.scala [hunk 3]
```diff
@@ -134,18 +193,42 @@ object Table {
...
+    globals: Row
+  ) = this(
+    hc,
+    ContextRDD.weaken(rdd),
+    signature,
+    key,
+    globalSignature,
+    globals)
+
+  def this(
```
Comment (catoverdrive) [inline src/main/scala/is/hail/table/Table.scala]: also here

**Comment by danking** [comment]
You and @jigold both with these "tests"

dr. evil making air quotes gif

**Comment by danking** [comment]
@catoverdrive back to you! did a rebase

**Pull request merged**